netops_demo/py/functions/generate/generate.py

```diff
@@ -135,20 +153,65 @@ def _generate_batch(context, start_timestamp, num_samples, interval):
     for sample_idx in range(num_samples):
         timestamp = int(start_timestamp + (interval * sample_idx))
 
-        for generated_metric_name, generated_metric in next(context.user_data.manager.generate()).items():
-
-            # get metric, or create one with the proper labels
-            metric_in_batch = metrics_batch.setdefault(generated_metric_name, {
-                'labels': context.user_data.configuration['metrics'][generated_metric_name].get('labels'),
-            })
-            
-            # shove values
-            metric_in_batch.setdefault('timestamps', []).append(timestamp)
-            metric_in_batch.setdefault('values', []).append(generated_metric['value'])
-            metric_in_batch.setdefault('alerts', []).append(generated_metric['alert'])
-            metric_in_batch.setdefault('is_error', []).append(1 if generated_metric['is_error'] else 0)
-
-    return metrics_batch
+        generated_metrics = next(context.user_data.deployment.generate())
+        for company, locations in generated_metrics.items():
+            # Get or Create company
+            dict_cmp = metrics_batch.setdefault(company, {})
+            for location, devices in locations.items():
+                current_coordinates = devices['location']
+                latitude = current_coordinates[0],
+                longtitude = current_coordinates[1]
+
+                # Get or create initial location labels
+                loc_labels = {
+                    'labels': {
+                        'latitude': latitude[0],
+                        'longtitude': longtitude,
+                        'name': company,
+                        'location': f'{company}/{location}'
+                    }
+                }
+                # Get or create location (company -> location)
+                dict_loc = dict_cmp.setdefault(location, {})
+
+                for device, metrics in devices['devices'].items():
```

Comment:
I would break this up into smaller functions - function became way too long

Reply:
Done